src/Sentry/Integrations/SystemDiagnosticsMetricsIntegration.cs

```diff
@@ -0,0 +1,83 @@
+#if NET8_0_OR_GREATER
+using System.Diagnostics.Metrics;
+using Sentry.Extensibility;
+
+namespace Sentry.Integrations;
+
+internal class SystemDiagnosticsMetricsIntegration : ISdkIntegration
+{
+    private static MeterListener? SentryMeterListener;
+    private static readonly object InitLock = new();
+
+    public void Register(IHub hub, SentryOptions options)
+    {
+        var listeners = options.ExperimentalMetrics?.SystemDiagnosticsMetricsListeners;
+        if (listeners is not { Count: > 0 })
+        {
+            options.Log(SentryLevel.Info,
+                "System.Diagnostics.Metrics Integration is disabled because no listeners configured.");
+            return;
+        }
+
+        if (SentryMeterListener is not null)
+        {
+            options.Log(SentryLevel.Info,
+                "System.Diagnostics.Metrics Integration has already been registered.");
+            return;
+        }
+
+        lock (InitLock)
```

Comment:
Nothing in Init is thread safe, do we need this?

Reply:
I was wary about initializing it twice and double counting metrics... but yeah, as it's just about static initialization I'll switch out the double lock for a Lazy<T>.